src/sentry/auth/services/auth/impl.py

```diff
@@ -26,6 +26,12 @@
 from sentry.silo.safety import unguarded_write
 from sentry.users.models.user import User
 
+# If querying for a provider, we also check for equivalent providers
+EQUIVALENT_PROVIDERS = {
+    "fly": ["fly", "fly-non-partner"],
+    "fly-non-partner": ["fly", "fly-non-partner"],
```

Comment:
Probably too late for this question 😅 but why did we have to create a different auth provider when someone changes the partnership plan?

I wonder if the right fix here is to delete all fly-non-partner providers and store someone not being on partner plan on a different data model.

Reply:
@sentaur-athena Good question! This is a remnant of [this project](https://www.notion.so/sentry/Partner-Billing-Migration-dc6719912e4943fbabe7173cfae1c295?pvs=4#ef4980b32aa94ddbb90ce313f5630208). We needed a way to make it so transitioned customers could disable the partner SSO, without making it that way for customers still on the partner plan. This was path caused the least friction, as we can't just conditionally modify the provider config dataclass. 

I guess another option was to add a new column to `AuthProvider`, ie. something to signal that the customer transitioned off partner, and check for that value when `is_partner` in the config is true, but this seemed like a lot of unnecessary work and would be useless for the majority of `AuthProvider` objects.